plane2/src/typed_socket/mod.rs

```diff
@@ -22,18 +22,18 @@ pub struct TypedSocket<T: ChannelMessage> {
     pub remote_handshake: Handshake,
 }
 
-pub struct TypedSocketSender<T: ChannelMessage> {
-    send: Sender<SocketAction<T>>,
+pub struct TypedSocketSender<A> {
+    inner_send: Box<dyn Fn(SocketAction<A>) -> anyhow::Result<()> + 'static + Send + Sync>,
```

Comment:
maybe have the error here as a parameter (like A, T: std::error::Error)

or a Box<dyn Error>

but this is ok for now.

Reply:
Done! I added a concrete error type.